### PR TITLE
pitr: support s3 object lock

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? (Required)

Add description about PITR supporting S3 Object Lock

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/16739
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: br/backup-and-restore-storages.md [hunk 1]
```diff
@@ -260,4 +260,6 @@ BR supports specifying the Azure server-side encryption scope or providing the e
 
 ## Other features supported by the storage service
 
-BR v6.3.0 supports AWS [S3 Object Lock](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock.html). You can enable this feature to prevent backup data from being tampered with or deleted.
+Amazon [S3 Object Lock](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock.html) can help prevent backup data from being accidentally or intentionally deleted during a specified retention period, enhancing the security and integrity of data. Starting from v6.3.0, BR supports Amazon S3 Object Lock in snapshot backups, adding an additional layer of security for full backups. Starting from v8.0.0, PITR also supports Amazon S3 Object Lock. Whether full backups or log data backups, the Object Lock feature ensures a more reliable data protection, further strengthening the security of data backup and recovery and meeting regulatory requirements.
```
[the remark this answers is not in the document]
Comment (lilin90) [reply br/backup-and-restore-storages.md:263]: We have been the "PITR" capitalization for a long time. Ref: https://docs.pingcap.com/tidb/stable/br-pitr-guide, https://en.wikipedia.org/wiki/Point-in-time_recovery

**Review by BornChanger — APPROVED** [review]

File: br/backup-and-restore-storages.md (hunk 1 above)
Comment (qiancai) [inline br/backup-and-restore-storages.md:263]:
```suggestion
Amazon [S3 Object Lock](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock.html) can help prevent backup data from accidental or intentional deletion during a specified retention period, enhancing the security and integrity of data. Starting from v6.3.0, BR supports Amazon S3 Object Lock for snapshot backups, adding an additional layer of security for full backups. Starting from v8.0.0, PITR also supports Amazon S3 Object Lock. Whether for full backups or log data backups, the Object Lock feature ensures more reliable data protection, further strengthening the security of data backup and recovery and meeting regulatory requirements.
```

File: br/backup-and-restore-storages.md [hunk 2]
```diff
@@ -260,4 +260,6 @@ BR supports specifying the Azure server-side encryption scope or providing the e
...
+
+BR and PITR automatically detect whether the Amazon S3 Object Lock feature is enabled or disabled. You do not need to make any additional changes.
```
Comment (qiancai) [inline br/backup-and-restore-storages.md:265]:
```suggestion
BR and PITR automatically detect whether the Amazon S3 Object Lock feature is enabled or disabled. You do not need to perform any additional operations.
```

Comment (qiancai) [reply br/backup-and-restore-storages.md:265]: The newly added warning in https://github.com/pingcap/docs-cn/pull/16739/files  is missing here.

Comment (lilin90) [reply br/backup-and-restore-storages.md:265]: > The newly added warning in https://github.com/pingcap/docs-cn/pull/16739/files is missing here.

Yes, I'll add the warning soon.

File: br/backup-and-restore-storages.md [hunk 3]
```diff
@@ -260,4 +260,6 @@ BR supports specifying the Azure server-side encryption scope or providing the e
 
 ## Other features supported by the storage service
 
-BR v6.3.0 supports AWS [S3 Object Lock](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock.html). You can enable this feature to prevent backup data from being tampered with or deleted.
+Amazon [S3 Object Lock](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock.html) can help prevent backup data from accidental or intentional deletion during a specified retention period, enhancing the security and integrity of data. Starting from v6.3.0, BR supports Amazon S3 Object Lock for snapshot backups, adding an additional layer of security for full backups. Starting from v8.0.0, PITR also supports Amazon S3 Object Lock. Whether for full backups or log data backups, the Object Lock feature ensures more reliable data protection, further strengthening the security of data backup and recovery and meeting regulatory requirements.
+
+BR and PITR automatically detect whether the Amazon S3 Object Lock feature is enabled or disabled. You do not need to perform any additional operations.
```
Comment (lilin90) [inline br/backup-and-restore-storages.md:265]:
```suggestion
BR and PITR automatically detect whether the Amazon S3 Object Lock feature is enabled or disabled. You do not need to perform any additional operations.

> **Warning:**
>
> If the Object Lock feature is enabled during the snapshot backup or PITR log backup process, the snapshot backup or log backup might fail. You need to restart the snapshot backup or PITR log backup task to continue the backup.
```

**Comment by lilin90** [comment]
@qiancai Thanks! Your comments have been resolved and I added the warning. PTAL again~

**Review by qiancai — APPROVED** [review]
LGTM

**Pull request merged**